internal/storetesting/entities.go

```diff
@@ -0,0 +1,102 @@
+// Copyright 2014 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package storetesting
+
+import (
+	jc "github.com/juju/testing/checkers"
+	gc "gopkg.in/check.v1"
+	"gopkg.in/juju/charm.v5-unstable"
+	"gopkg.in/mgo.v2"
+
+	"gopkg.in/juju/charmstore.v4/internal/mongodoc"
+)
+
+// EntityBuilder provides a convenient way to describe a mongodoc.Entity
+// for tests that is correctly formed and contain the desired
+// information.
+type EntityBuilder struct {
+	entity *mongodoc.Entity
+}
+
+// NewEntity creates a new EntityBuilder for the provided URL.
+func NewEntity(url string) EntityBuilder {
```

Comment:
I think I'm, going to leave these as they are I quite like that you can read it as NewEntity(...).WithPromulgatedURL(...)